File: library-infrastructure/src/main/java/com/dodecaedro/library/infrastructure/repository/ElasticBookSearchDocumentRepository.java
package com.dodecaedro.library.infrastructure.repository;

import com.dodecaedro.library.infrastructure.search.BookSearchDocument;
import org.springframework.data.elasticsearch.repository.ElasticsearchRepository;

import java.util.List;
import java.util.UUID;

public interface ElasticBookSearchDocumentRepository extends ElasticsearchRepository<BookSearchDocument, UUID> {
	List<BookSearchDocument> findByTitle(String title);
}

File: library-infrastructure/src/main/kotlin/com/dodecaedro/library/infrastructure/repository/ElasticBookSearchDocumentRepository.kt
package com.dodecaedro.library.infrastructure.repository

import com.dodecaedro.library.infrastructure.search.BookSearchDocument
import org.springframework.data.elasticsearch.repository.ElasticsearchRepository
import java.util.*

interface ElasticBookSearchDocumentRepository: ElasticsearchRepository<BookSearchDocument, UUID> {
	fun findByTitle(title: String): List<BookSearchDocument>
}
